### PR TITLE
decouple caching logic from read event

### PR DESCRIPTION
**Pull request by rex-god**

### CODE REVIEW

File: src/modules/helper/db-sync/nft-db-sync.service.ts [hunk 1]
```diff
@@ -101,44 +102,48 @@ export class NFtDbSyncService {
...
+  }
   
   private async prepareNftDbDto(event: NftReadEvent): Promise<NftInfo> {
     try {
       const nftData = await this.remoteDataFetcher.fetchNftDetails(new FetchNftDto(event.network, event.tokenAddress));
       const nftDbDto = nftData.getNftInfoDto();
-      const oldCachedUri = (await this.nftInfoAccessor.findOne({ network: event.network, mint: event.tokenAddress }))?.cached_image_uri;
-      const cachedImageUri = await this.cacheNftData(new NftCacheEvent(oldCachedUri, nftDbDto.image_uri));
-      nftDbDto.cached_image_uri = cachedImageUri;
+      nftDbDto.cached_image_uri = nftDbDto.image_uri;
```
Comment (codinpsycho) [inline src/modules/helper/db-sync/nft-db-sync.service.ts:146]: Wont this rewrite cached_image_uri in DB?

Comment (rex-god) [reply src/modules/helper/db-sync/nft-db-sync.service.ts]: removed

File: src/common/utils/utils.ts [hunk 2]
```diff
@@ -161,12 +161,12 @@ export const Utility = {
...
-  isUriMatched: (cachedImageUri: string, imageUri: string): boolean => {
-    if (!cachedImageUri) return false;
-    const cdnUri = cachedImageUri.split('/').slice(4).join('/');
+  isUriMatched: (ipfsFileUri: string, cachedFileUri?: string): boolean => {
+    if (!cachedFileUri) return false;
+    const cdnUri = cachedFileUri.split('/').slice(4).join('/');
     const ext = cdnUri.split('.').pop();
     const cdnUriWithoutExt = cdnUri.replace(`.${ext}`, '');   
-    if (cdnUriWithoutExt === imageUri) {
+    if (cdnUriWithoutExt === ipfsFileUri) {
```
Comment (codinpsycho) [inline src/common/utils/utils.ts:169]: this can just be simplified to return cdnUriWithoutExt === ipfsFileUri

File: src/modules/helper/db-sync/nft-db-sync.service.ts [hunk 3]
```diff
@@ -101,44 +107,48 @@ export class NFtDbSyncService {
...
     } catch (error) {
       console.error(error);
     }
   }
+
+  private async syncNftData(event: NftReadEvent): Promise<NftInfoDocument> {
+    try {
+      const nftDbDto = await this.prepareNftDbDto(event);
+      const result = await this.nftInfoAccessor.updateNft(nftDbDto);
+      const oldCachedUri = (await this.nftInfoAccessor.findOne({ network: event.network, mint: event.tokenAddress }))?.cached_image_uri;
```
Comment (codinpsycho) [inline src/modules/helper/db-sync/nft-db-sync.service.ts:139]: Why is this required? You are doing findAndUpdate with new : true. You can just use the result from updateNft() Same as issuing a find again

File: src/modules/helper/db-sync/nft-db-sync.service.ts [hunk 4]
```diff
@@ -8,12 +9,13 @@ import { RemoteDataFetcherService } from '../remote-data-fetcher/data-fetcher.se
 import { FetchNftDto, FetchAllNftDto, FetchAllNftByCreatorDto } from '../remote-data-fetcher/dto/data-fetcher.dto';
 import { NftCacheEvent, NftCreationEvent, NftDeleteEvent, NftReadByCreatorEvent, NftReadEvent, NftReadInWalletEvent, NftUpdateEvent } from './db.events';
 
-const afterNftCreationWaitTime_ms = 7000;
-const afterNftUpdateWaitTime_ms = 7000;
+const afterNftCreationWaitTime_ms = 15000;
```
Comment (codinpsycho) [inline src/modules/helper/db-sync/nft-db-sync.service.ts:12]: Why 15 secs?

Comment (rex-god) [reply src/modules/helper/db-sync/nft-db-sync.service.ts:12]: 7 seconds is too early for fetching data from blockchain.

File: src/modules/helper/db-sync/nft-db-sync.service.ts (hunk 3 above)
Comment (rex-god) [reply src/modules/helper/db-sync/nft-db-sync.service.ts:139]: Need to change.

File: src/common/utils/utils.ts (hunk 2 above)
Comment (rex-god) [reply src/common/utils/utils.ts:169]: Ok

**Review by codinpsycho — APPROVED** [review]
